pkg/resources/block_storage/resource_volume.go

```diff
@@ -0,0 +1,588 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	exoscale "github.com/exoscale/egoscale/v3"
+
+	"github.com/exoscale/terraform-provider-exoscale/pkg/config"
+	providerConfig "github.com/exoscale/terraform-provider-exoscale/pkg/provider/config"
+	"github.com/exoscale/terraform-provider-exoscale/pkg/utils"
+)
+
+const ResourceVolumeDescription = `Manage [Exoscale Block Storage](https://community.exoscale.com/documentation/block-storage/) Volume.
+
+Block Storage offers persistent externally attached volumes for your workloads.
+`
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &ResourceVolume{}
+var _ resource.ResourceWithImportState = &ResourceVolume{}
+
+// ResourceVolume defines the resource implementation.
+type ResourceVolume struct {
+	client *exoscale.Client
+}
+
+// NewResourceVolume creates instance of ResourceVolume.
+func NewResourceVolume() resource.Resource {
+	return &ResourceVolume{}
+}
+
+// ResourceVolumeModel defines the resource data model.
+type ResourceVolumeModel struct {
+	ID             types.String `tfsdk:"id"`
+	Name           types.String `tfsdk:"name"`
+	Size           types.Int64  `tfsdk:"size"`
+	Blocksize      types.Int64  `tfsdk:"blocksize"`
+	CreatedAt      types.String `tfsdk:"created_at"`
+	Labels         types.Map    `tfsdk:"labels"`
+	SnapshotTarget types.Object `tfsdk:"snapshot_target"`
+	State          types.String `tfsdk:"state"`
+	Zone           types.String `tfsdk:"zone"`
+
+	Timeouts timeouts.Value `tfsdk:"timeouts"`
+}
+
+// Metadata specifies resource name.
+func (r *ResourceVolume) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_block_storage_volume"
+}
+
+// Schema defines resource attributes.
+func (r *ResourceVolume) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: ResourceVolumeDescription,
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "The ID of this resource.",
+				Computed:            true,
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "❗ Volume name.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"size": schema.Int64Attribute{
+				MarkdownDescription: "Volume size in GB. If volume is attached, instance must be stopped to update this value. Volume can only grow, cannot be shrunk.",
```

Comment:
Sure but description is still true. I can expand description to suggest both options:
- shutdown instance & resize volume;
- detach volume, resize it and re-attach it.

Not sure it is smart to do it automatically (detach -> resize -> attach) it will certainly break whatever workflow runs in the vm.

Reply:
Yes, we can expand the description, yes auto detach and attach could break some customer workload, I don't think it's convenient here you are right

Reply:
Fixed in https://github.com/exoscale/terraform-provider-exoscale/pull/341/commits/628800cbaaa8d94234f2577e09cc3fc315772940